internal/pkg/chainsync/chainsync.go

```diff
@@ -16,6 +16,7 @@ type BlockProposer interface {
 	SendHello(ci *block.ChainInfo) error
 	SendOwnBlock(ci *block.ChainInfo) error
 	SendGossipBlock(ci *block.ChainInfo) error
+	WaitForTarget(ci *block.ChainInfo) func()
```

Comment:
I think `WaiterForTarget` would be slightly better, since this call doesn't block.